### PR TITLE
Run int tests only in PRs

### PR DESCRIPTION
**Pull request by psfinaki**

### CODE REVIEW

**Review by T-Gro — APPROVED** [review]

File: eng/Build.ps1 [hunk 1]
```diff
@@ -620,6 +620,10 @@ try {
         TestUsingNUnit -testProject "$RepoRoot\vsintegration\tests\UnitTests\VisualFSharp.UnitTests.fsproj" -targetFramework $desktopTargetFramework -testadapterpath "$ArtifactsDir\bin\VisualFSharp.UnitTests\"
         TestUsingXUnit -testProject "$RepoRoot\vsintegration\tests\FSharp.Editor.Tests\FSharp.Editor.Tests.fsproj" -targetFramework $desktopTargetFramework -testadapterpath "$ArtifactsDir\bin\FSharp.Editor.Tests\FSharp.Editor.Tests.fsproj"
     }
+    
+    if ($testIntegration) {
+        TestUsingXUnit -testProject "$RepoRoot\vsintegration\tests\FSharp.Editor.IntegrationTests\FSharp.Editor.IntegrationTests.csproj" -targetFramework $desktopTargetFramework -testadapterpath "$ArtifactsDir\bin\FSharp.Editor.IntegrationTests\"
```
Comment (vzarytovskii) [inline eng/Build.ps1:625]: Won't this still run them in every pipeline, whenre we do `testAll`?

Comment (psfinaki) [reply eng/Build.ps1:635]: Right, thanks. Let's have a separate flag then.

File: eng/Build.ps1 [hunk 2]
```diff
@@ -148,8 +149,17 @@ function Process-Arguments() {
         $script:testVs = $True
     }
 
+    if ($testAllButIntegration) {
```
Comment (vzarytovskii) [inline eng/Build.ps1:152]: That looks a bit clunky, we probably should just have a separate leg, which runs with `testIntegration` only.

Comment (psfinaki) [reply eng/Build.ps1:153]: So alternatively we can just set the value to false [here](https://github.com/dotnet/fsharp/blob/main/eng/Build.ps1#LL147-L147C40) but that will make the name "-test**All**" a lie. I wouldn't rename it to "-testAllUnit" either as, well, not many of those tests are unit tests. Hence this new switch... But I don't have a very strong stance on this, just making a point.

**Review by KevinRansom — APPROVED** [review]

File: eng/Build.ps1 (hunk 2 above)
Comment (KevinRansom) [reply eng/Build.ps1:153]: I'm okay with the test framework having explicit subsets of tests.  We do it a lot for main test suites.  I.e. testCambridge, --testFSharpQa etc.

**Pull request merged**